toolchain/check/facet_type.cpp

```diff
@@ -62,12 +62,10 @@ static auto IncompleteFacetTypeDiagnosticBuilder(
     return context.emitter().Build(loc_id, ImplAsIncompleteFacetTypeDefinition,
                                    facet_type_inst_id);
   } else {
-    CARBON_DIAGNOSTIC(
-        ImplAsIncompleteFacetTypeRewrites, Error,
-        "declaration of impl as incomplete facet type {0} with rewrites",
-        InstIdAsType);
-    return context.emitter().Build(loc_id, ImplAsIncompleteFacetTypeRewrites,
-                                   facet_type_inst_id);
+    context.TODO(loc_id,
+                 "declaration of impl as incomplete interface with a rewrite "
+                 "constraint");
```

Comment:
I am quite convinced this case should no longer ever happen so I changed it into a CHECK with a TODO to remove the parameter. I tried to organize the type completion of the facet type in `impl as` a bit better since it occurs in two different places with some loose coordination right now, but it's very tricky. I will try again elsewhere.